llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -4237,6 +4242,81 @@ getFalkorUnrollingPreferences(Loop *L, ScalarEvolution &SE,
   }
 }
 
+static bool shouldUnrollLoopWithInstruction(Instruction &I,
+                                            AArch64TTIImpl &TTI) {
+  // Don't unroll vectorised loop.
+  if (I.getType()->isVectorTy())
+    return false;
+
+  if (isa<CallInst>(I) || isa<InvokeInst>(I)) {
```

Comment:
Since we are only working with `CallBase` below, we can rewrite it as:
```
if (auto *CB = dyn_cast<CallBase>(&I)) {
  if (const Function *F = CB->getCalledFunction())
     ....
}
```

Reply:
Hmm, it's a good suggestion but I checked and it seems `CallBrInst` also derives from `CallBase`. I'm not sure if I should implicitly add support for that by permitting all instances of `CallBase`. If you don't mind I'll leave this as it is for now?

Reply:
Yep, you're right! I've fixed it now.